src/material.cpp

```diff
@@ -221,19 +227,45 @@ void pyne::Material::from_hdf5(std::string filename, std::string datapath, int r
   // Open the database
   hid_t db = H5Fopen(filename.c_str(), H5F_ACC_RDONLY, fapl);
 
-  bool datapath_exists = h5wrap::path_exists(db, datapath);
-  if (!datapath_exists)
-    throw h5wrap::PathNotFound(filename, datapath);
 
   // Clear current content
   comp.clear();
-
   // Load via various protocols
-  if (protocol == 0)
+  if (protocol == 0) {
+    bool datapath_exists = h5wrap::path_exists(db, datapath);
+    if (!datapath_exists)
+      throw h5wrap::PathNotFound(filename, datapath);
     _load_comp_protocol0(db, datapath, row);
-  else if (protocol == 1)
-    _load_comp_protocol1(db, datapath, row);
-  else
+  } else if (protocol == 1) {
+    
+    // Check /material type
+    // set test variables
+    herr_t status= H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+    H5O_info_t object_info;
+    
+    // check "/matierial" path exist as a non-dataset
+    status = H5Oget_info_by_name(db, "/material" , &object_info, H5P_DEFAULT);
+    bool material_notdataset_exists = (status == 0 && object_info.type != H5O_TYPE_DATASET);
+   
+    // Reset status and test for datapath
+    status = H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+    status = H5Oget_info_by_name(db, datapath.c_str(), &object_info, H5P_DEFAULT);
+    bool datapath_notdataset_exists = (status == 0 && object_info.type != H5O_TYPE_DATASET);
+    
+    // Group "/material" does not exist as a non-dataset
+    if (!material_notdataset_exists || !datapath_notdataset_exists) {
```

Comment:
@gonuke does it make more sense ?
I used your suggestion, renamed the `bool` variable so it contains all the info and added some comments

Reply:
I think the logic is clearer, but there is some lingering confusion (maybe it's just me) because we are mixing a few cases here.

It is old format if either "/material" or `datapath` is a dataset  - this is clear
It is new format if "/material" is a groupset   - this is clear
It is irrelevant if datapath is a non-dataset   - this is implied but not clear

So maybe some new variable names/logic (pseudocode):

```
enum data_layout_types = {unknown, old_format, new_format};

data_layout_types this_data_layout = unknown

// get info for `datapath`
if (status != 0 || object_info.type == DATASET) this_data_layout = old_format

// get info for "/material"
if (status != 0 || object_info.type == DATASET) { 
    this_data_layout = old_format
} else if (status == 0 && object_info.type == GROUP) {
    this_data_layout = new_format
}

```

Reply:
instead of the `enum` you could have two `bool`: `format_detected` and `new_format`